### PR TITLE
fix: schema

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Original schema specifies `cuid`, which is problematic on 2 fronts
- first, cuid is a randomly generated id which is difficult for our users to remember 
- next, cuid is not generated in db (prisma uses js to autogen). this leads to typing on kysely output being `String` instead of `GeneratedAlways<String>`.

## Solution
- change to auto-incrementing ids
    - note that now, the output of kysely types is `GeneratedAlways`
- update migration

### CODE REVIEW

File: apps/studio/src/server/modules/resource/resource.types.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
-export interface NavbarItem {
-  name: string
+import { type IsomerSiteProps } from '@opengovsg/isomer-components'
+
+export type Navbar = { items: IsomerSiteProps['navBarItems'] }
+
+export interface FooterItem {
+  title: string
```
Comment (seaerchin) [inline apps/studio/src/server/modules/resource/resource.types.ts:6]: this is cherry-picked, downstream will refer correctly to `IsomerSiteProps`

File: apps/studio/prisma/schema.prisma [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// This is your Prisma schema file,
+// This is your Prisma schema file,/
```
Comment (dcshzj) [inline apps/studio/prisma/schema.prisma:1]: nit: extra slash

File: apps/studio/src/server/modules/resource/resource.types.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
-export interface NavbarItem {
-  name: string
+import { type IsomerSiteProps } from '@opengovsg/isomer-components'
+
+export type Navbar = { items: IsomerSiteProps['navBarItems'] }
+
+export interface FooterItem {
```
Comment (dcshzj) [inline apps/studio/src/server/modules/resource/resource.types.ts:5]: nit: I suspect most of the types that we need on the CMS should be able to be derived from the exported props one way or another, so we might not need to re-define them here.

File: apps/studio/prisma/seed.ts [hunk 4]
```diff
@@ -86,13 +86,8 @@ async function main() {
...
     .values({
-      id: '1',
-      siteId: '1',
-      content: {
-        name: 'navi',
-        url: 'www.isomer.gov.sg',
-        items: NAV_BAR_ITEMS,
-      } satisfies Navbar,
+      siteId: id,
+      content: { items: NAV_BAR_ITEMS } satisfies Navbar,
```
Comment (dcshzj) [inline apps/studio/prisma/seed.ts:90]: question(non-blocking): are we able to type `NAV_BAR_ITEMS` to `Navbar` directly when we define it above in line 14?

**Review by dcshzj — APPROVED** [review]
No blocking comments

File: apps/studio/src/server/modules/resource/resource.types.ts (hunk 3 above)
Comment (seaerchin) [reply apps/studio/src/server/modules/resource/resource.types.ts:5]: exported for ease of use cos it's very tedious to type `Isomer...['...']` every time ._.

File: apps/studio/prisma/schema.prisma (hunk 2 above)
Comment (seaerchin) [reply apps/studio/prisma/schema.prisma:1]: idk how that got there but thanks for the catch

File: apps/studio/prisma/seed.ts (hunk 4 above)
Comment (seaerchin) [reply apps/studio/prisma/seed.ts:90]: added on otp, still kept check here

File: apps/studio/prisma/generated/generatedTypes.ts [hunk 5]
```diff
@@ -8,43 +8,43 @@ export type Timestamp = ColumnType<Date, Date | string, Date | string>
...
-  resourceId: string
-  userId: string
+  id: GeneratedAlways<number>
+  resourceId: number
+  userId: number
   role: RoleType
 }
 export type Resource = {
-  id: string
+  id: GeneratedAlways<number>
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply apps/studio/prisma/generated/generatedTypes.ts:31]: yes, they've been updated here

**Pull request merged**